Updated combined program

deswa7310 opened the pull request:
Literally spaghetti. Contains code from all 3 programs. Should work for all 3 mazes, and if it doesn't then I'm going to cry. I've tested it but you guys should too.

deswa7310 committed:
Updated combined program

Literally spaghetti. Contains code from all 3 programs. Should work for all 3 mazes, and if it doesn't then I'm going to cry. I've tested it but you guys should too.

deswa7310 commented:
I've counted and there's only around 44 semicolons in the main function, so we won't get marked down for it.

Kaolinnite reviewed (approved):
Good job on the integration and challenge especially. All that's left is fully optimize the code and add additional features

Kaolinnite committed:
Update robot.cpp

- Added return lines so the robot stops after encountering the flag; removed some unnecessary lines
- Cleaned up some of the code and documentation

deswa7310 committed:
Merge pull request #31 from deswa7310/Kaolinnite-patch-1

Final code cleanup

deswa7310 committed:
Final code clean-up and minor addition

 - Added plenty more comments and built upon Tristan's ideas/edits. 
 - Idea of killing the program once the robot detected the finish flag was really good, and I added a delay so it stops once it reaches the flag (not before it). 
 - "containsBlack()" method got scarily close to detecting the black pixels off the side of the image, so I made it scan row 75 instead of 50 just as a precaution (walls might be closer to edge of image in marker's challenge maze). 
 - Also added a small "Finished :)" message before program is killed. 
 - Edited some booleans to make them tidier/more efficient. 
 - Added a semi colon count at the top of main (our program looks massive but main actually only has around 40 semicolons, it's mainly just whitespace and if statements haha). 
 - Edited some comments to make them more accurate (some were outdated).

Kaolinnite committed:
Update robot.cpp

Changes:
- Edited the documentation lines
- Moved some lines elsewhere; renamed some functions
- Removed the delay function; robot over shot once it reached flag. Change to detection position seemed to be sufficient for stopping the robot
- Edited the message slightly

I can't really think of any other adjustments other than creating a class or struct, but that may require a complete overhaul of the architecture, but as always feel free to submit your own changes

deswa7310 committed:
Merge pull request #33 from deswa7310/Kaolinnite-patch-2

Update robot.cpp

deswa7310 committed:
Update robot.cpp

Added some more minor documentation edits.
- Updated semicolon count.
- Made vGo static too.
- Updated some comments.
- Note: methods and functions are slightly different. Methods act on the data within objects passed to them (Java only uses methods) whereas functions just act on the data passed to them. Since we aren't using objects or classes, we are only using functions.

Jayman874 committed:
Merge pull request #34 from deswa7310/deswa7310-patch-2

Update robot.cpp

Kaolinnite committed:
Merge pull request #32 from deswa7310/deswa7310-patch-1

Final code clean-up and minor addition

Kaolinnite commented:
All edits on this branch finalized, tested, and approved . Merging to main branch now
